### ISSUE TITLE
Questions from Brian Ruttenburg

### ISSUE THREAD

**Issue by probinso** [GaloisInc/ppaml-cp6#3]
Hi Galois PPAML team,
 
We had a couple of quick questions from reading the CP6 description. We’re hoping you can clarify:
 

* In the probability model section it talks about the Un,m(x) as the pairwise potential between two images. It says “It refers to the features that co-exist for the two images, i.e. [Hn(x) Hm(x)]”. Is this notation referring to the dot product between Hn and Hm or a vector concatenation?

* Are the queries used for evaluation only single value (e.g., MAP) queries? Will there be a need to retrieve the posterior distribution over labels for the test images?

 
Thanks!
 
Brian

**Comment by eswears** [comment]
Brian,

 

Thanks for the questions.  I see three questions in your e-mail, but four numbers in the list, so I’ll answer the three that I see and just let me know if I missed a fourth one:

 

1-      The hn(x) and hm(x) variables are represented as a 1 x (C+1) vectors.  These are then concatenated to form [hn(x) hm(x)] which is 1 x 2*(c+1).  You are welcome to experiment with other pairwise potentials.

2-      If I follow your second question about queries correctly, you’re asking if each hidden node (one node per image) is only assigned a single label when classifying the test data (e.g. 0 or 1, where “0” means the current label does not get assigned to that image and “1” means it does).  If so, then yes.  Each hidden nodes has two states when classifying on a per label basis, “0” or “1”, but you will need to run your classifiers for each label class, which will results in a single image having multiple class labels.  This was done to keep the model simple and more efficient, but one could also create a CRF that has 25 states one for each label class and one for the “other” class.

3-      To do the evaluation we will need the probability that each label exists in each image. No posterior calculations over the labels are required for this, since the CRFs and their label probabilities are independent of the other labels. However, one could calculate the posteriors given the prior probabilities of the labels in the training data.

 

Hope this helps,

Eran

**Comment by bruttenberg** [comment]
Hi Eran, thanks for the responses.

I don't think answers 2 and 3 were getting at the heart of my question. Let's assume a single label right now, and we have a vector of hidden nodes that indicates whether image i has this label. To do the evaluation, are you looking for a single state of the vector (e.g., the MAP state of the vector, a classification decision for each image) or a distribution over that vector (e.g., a set of vectors with associated probabilities)? And yes, I understand that we would have a separate answer for each label since they are independent.

Does my question make sense?

**Comment by eswears** [comment]
Thanks for the clarification.

 

This problem is formulated different than most CRF based image segmentation problems. So, instead of having one hidden node per pixel/grid-cell that form a vector of hidden nodes for a single image, as is done for image segmentation, we step back a level and have one hidden node per image over a database of images. So, the vector of hidden nodes is no longer over a single image, it is over the database of images, one label per image.

 

The evaluation will require one state for each hidden node, so one classification decision for each image.

 

Hope this helps.

 

-Eran

**Comment by bruttenberg** [comment]
It does, thanks!

**Comment by collinsr** [comment]
In particular, we're expecting a per-image vector of label likelihoods (the
label_vector in the Image Table) along with a single vector of thresholds,
one entry per label, for converting those per-image likelihoods into binary
yes/no decisions.  (See 'Test Output Format' on page 12 of the problem
statement for details.)

-Roddy

**Comment by cscherrer** [comment]
@eswears Has documentation been updated to clarify this? We'll close this issue once we're notified.

**Comment by collinsr** [comment]
I believe the problem statement as of v18 (mid-July, current on git)
clarifies this (see "Test Output Format" on page 12.) Let me know if it's
insufficiently clear.

Thanks,
-Roddy

**Comment by eswears** [comment]
I’m adding some more text now to make it a little more clear.

 

-Eran

**Comment by eswears** [comment]
I just pushed version 20 of the problem statement, which adds more details from the answers that I gave for Brian’s questions. Let us know if you believe further clarification is needed.

 

Thanks,

Eran 

 

 

 

From: Eran Swears [mailto:eran.swears@kitware.com] 
Sent: Tuesday, September 15, 2015 1:47 PM
To: 'GaloisInc/ppaml-cp6'; 'GaloisInc/ppaml-cp6'
Subject: RE: [ppaml-cp6] Questions from Brian Ruttenburg (#3)

 

I’m adding some more text now to make it a little more clear.

 

-Eran
